File: selfstudy/js/books_sortdesc.js
/*
task 60:
Books - SortDesc
You need to sort the values in descending order for book page counts
*/

function sort_down_books(pages) {
  return [...pages].sort((a, b) => b - a);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: sort_down_books([10, 20, 30]) },
  { input: [50, 60, 70], expected: sort_down_books([50, 60, 70]) },
  { input: [1, 2, 3], expected: sort_down_books([1, 2, 3]) },
  { input: [100, 200], expected: sort_down_books([100, 200]) },
  { input: [], expected: sort_down_books([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('sort_down_books', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(sort_down_books(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, sort_down_books(t.input) === t.expected || JSON.stringify(sort_down_books(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think

File: selfstudy/python/books_sortdesc.py
'''
task 160:
Books - SortDesc
You need to sort the values in descending order for book page counts
'''

def sort_down_books(pages):
    return sorted(pages, reverse=True)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], sort_down_books([10, 20, 30])),
        ([50, 60, 70], sort_down_books([50, 60, 70])),
        ([1, 2, 3], sort_down_books([1, 2, 3])),
        ([100, 200], sort_down_books([100, 200])),
        ([], sort_down_books([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = sort_down_books(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# finally works
